### PR TITLE
apache-ctakes: fix url, deprecate

### PR DESCRIPTION
**Pull request by iMichka**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by iMichka** [comment]
The checksum has changed. I downloaded it from https://ctakes.apache.org/downloads.cgi.
Upstream might have redone their release while moving to the new place.
I'm rev-bumping this, in the case there are any changes that need to be shipped to our users (I don't have the original source file for comparison).

**Comment by iMichka** [comment]
Actually it is expected that the sha256 changes, as we are using the source tarball now.

**Comment by iMichka** [comment]
This failed to build with openjdk, so I switched back to openjdk 8.
The build did make some progress but now it fails by downloading some stuff:

```
2021-12-06T20:24:02.3983200Z [ERROR] Failed to execute goal on project ctakes-ytex-web: Could not resolve dependencies for project org.apache.ctakes:ctakes-ytex-web:war:4.0.0.1: Failed to collect dependencies at org.eclipse.jetty:jetty-runner:jar:9.1.0.M0 -> org.eclipse.jetty:jetty-jsp:jar:9.1.0.M0 -> org.glassfish.web:javax.servlet.jsp:jar:2.3.2 -> org.glassfish:javax.el:jar:3.0.1-b06-SNAPSHOT: Failed to read artifact descriptor for org.glassfish:javax.el:jar:3.0.1-b06-SNAPSHOT: Could not transfer artifact org.glassfish:javax.el:pom:3.0.1-b06-SNAPSHOT from/to maven-default-http-blocker (http://0.0.0.0/): Blocked mirror for repositories: [apache.snapshots (http://repository.apache.org/snapshots, default, snapshots), sonatype-snapshots (http://oss.sonatype.org/content/groups/jetty, default, releases+snapshots)] -> [Help 1]
2021-12-06T20:24:02.3993840Z org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal on project ctakes-ytex-web: Could not resolve dependencies for project org.apache.ctakes:ctakes-ytex-web:war:4.0.0.1: Failed to collect dependencies at org.eclipse.jetty:jetty-runner:jar:9.1.0.M0 -> org.eclipse.jetty:jetty-jsp:jar:9.1.0.M0 -> org.glassfish.web:javax.servlet.jsp:jar:2.3.2 -> org.glassfish:javax.el:jar:3.0.1-b06-SNAPSHOT
```

**Comment by SMillerDev** [comment]
How about openjdk@11?

**Comment by iMichka** [comment]
I'll give it another try later today. If it does not work; I'll open an issue upstream.

**Comment by iMichka** [comment]
```
constituent[40]: file:/opt/homebrew/Cellar/maven/3.8.4/libexec/lib/guava-25.1-android.jar
---------------------------------------------------
Exception in thread "main" java.lang.Error: java.lang.reflect.InvocationTargetException
	at com.sun.tools.xjc.reader.Ring.get(Ring.java:113)
	at com.sun.tools.xjc.reader.xmlschema.BGMBuilder.<init>(BGMBuilder.java:147)
	at com.sun.tools.xjc.reader.xmlschema.BGMBuilder.build(BGMBuilder.java:117)
	at com.sun.tools.xjc.ModelLoader.annotateXMLSchema(ModelLoader.java:425)
	at com.sun.tools.xjc.ModelLoader.load(ModelLoader.java:174)
	at com.sun.tools.xjc.ModelLoader.load(ModelLoader.java:119)
	at com.sun.tools.xjc.Driver.run(Driver.java:333)
	at org.codehaus.mojo.jaxb2.AbstractXjcMojo.execute(AbstractXjcMojo.java:316)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:972)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:293)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:196)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at com.sun.tools.xjc.reader.Ring.get(Ring.java:102)
	... 29 more
Caused by: java.lang.Error: java.lang.reflect.InvocationTargetException
	at com.sun.tools.xjc.reader.Ring.get(Ring.java:113)
	at com.sun.tools.xjc.reader.xmlschema.BindingComponent.getClassSelector(BindingComponent.java:65)
	at com.sun.tools.xjc.reader.xmlschema.ColorBinder.<init>(ColorBinder.java:62)
	at com.sun.tools.xjc.reader.xmlschema.BindGreen.<init>(BindGreen.java:63)
	... 34 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at com.sun.tools.xjc.reader.Ring.get(Ring.java:102)
	... 37 more
Caused by: java.lang.NoClassDefFoundError: javax/activation/MimeTypeParseException
	at java.base/java.lang.Class.getDeclaredConstructors0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredConstructors(Class.java:3137)
	at java.base/java.lang.Class.getConstructor0(Class.java:3342)
	at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2553)
	at com.sun.tools.xjc.reader.Ring.get(Ring.java:100)
	at com.sun.tools.xjc.reader.xmlschema.DefaultClassBinder.<init>(DefaultClassBinder.java:98)
	at com.sun.tools.xjc.reader.xmlschema.ClassSelector.<init>(ClassSelector.java:214)
	... 42 more
Caused by: java.lang.ClassNotFoundException: javax.activation.MimeTypeParseException
	at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass(SelfFirstStrategy.java:50)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.unsynchronizedLoadClass(ClassRealm.java:271)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:247)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:239)
	... 49 more
```

Here is the failure with `openjdk@11`, which is sign that it needs `opendk@8`

**Comment by iMichka** [comment]
I opened an issue upstream: https://issues.apache.org/jira/browse/CTAKES-556

File: Formula/apache-ctakes.rb [hunk 1]
```diff
@@ -1,8 +1,9 @@
 class ApacheCtakes < Formula
   desc "NLP system for extraction of information from EMR clinical text"
   homepage "https://ctakes.apache.org"
-  url "https://apache.claz.org/ctakes/ctakes-4.0.0.1/apache-ctakes-4.0.0.1-bin.tar.gz"
-  sha256 "f741016e3755054876f3bb27f916a8008af27175ef33785638a6292d300c972e"
+  url "https://dlcdn.apache.org//ctakes/ctakes-4.0.0.1/apache-ctakes-4.0.0.1-src.tar.gz"
+  sha256 "a4e0eb3094bc8dc134487548d4f2b49aa7b8cda41587e4300f34d64179f78271"
+  revision 1
```
Comment (chenrui333) [inline Formula/apache-ctakes.rb:6]: We can also add license in here
```suggestion
  license "Apache-2.0"
  revision 1
```

license ref, https://svn.apache.org/repos/asf/ctakes/trunk/LICENSE

**Comment by iMichka** [comment]
> Error: Empty installation

I don't know how to tell maven to install stuff. I just followed the upstream build instructions but that does not seem to be enough.

File: Formula/apache-ctakes.rb [hunk 2]
```diff
@@ -1,8 +1,10 @@
 class ApacheCtakes < Formula
   desc "NLP system for extraction of information from EMR clinical text"
   homepage "https://ctakes.apache.org"
-  url "https://apache.claz.org/ctakes/ctakes-4.0.0.1/apache-ctakes-4.0.0.1-bin.tar.gz"
+  url "https://dlcdn.apache.org//ctakes/ctakes-4.0.0.1/apache-ctakes-4.0.0.1-bin.tar.gz"
   sha256 "f741016e3755054876f3bb27f916a8008af27175ef33785638a6292d300c972e"
+  license "Apache-2.0"
+  revision 1
```
Comment (SMillerDev) [inline Formula/apache-ctakes.rb:7]: If we're only deprecating it, this should probably be:
```suggestion
```

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**